subscriber_notifee.go

```diff
@@ -117,39 +111,20 @@ func (nn *subscriberNotifee) subscribe(proc goprocess.Process) {
 	}
 }
 
-func handlePeerIdentificationCompletedEvent(dht *IpfsDHT, e event.EvtPeerIdentificationCompleted) {
-	dht.plk.Lock()
-	defer dht.plk.Unlock()
-	if dht.host.Network().Connectedness(e.Peer) != network.Connected {
-		return
-	}
-
-	// if the peer supports the DHT protocol, add it to our RT and kick a refresh if needed
-	valid, err := dht.validRTPeer(e.Peer)
-	if err != nil {
-		logger.Errorf("could not check peerstore for protocol support: err: %s", err)
-		return
-	}
-	if valid {
-		dht.peerFound(dht.ctx, e.Peer)
-		fixLowPeers(dht)
-	}
-}
-
 func handlePeerProtocolsUpdatedEvent(dht *IpfsDHT, e event.EvtPeerProtocolsUpdated) {
 	valid, err := dht.validRTPeer(e.Peer)
 	if err != nil {
 		logger.Errorf("could not check peerstore for protocol support: err: %s", err)
 		return
 	}
 
-	if valid {
-		dht.peerFound(dht.ctx, e.Peer)
-	} else {
+	if !valid {
 		dht.peerStoppedDHT(dht.ctx, e.Peer)
+	} else {
+		// we just might have discovered a peer that supports the DHT protocol
+		dht.fixLowPeers()
```

Comment:
Why not always call `fixLowPeers`? It won't do anything if we have peers in the routing table, but might help if we're low on peers. Do you think we'll run into startup issues (eating lots of resources) from doing too many self queries?

Reply:
@aschmahmann We are in a ways ALWAYS calling i here as the `peerStoppedDHT` function also calls `fixLowPeers`.

I've added a return to the if branch and removed the else.